harness/tests/storage/test_shared_fs.py

```diff
@@ -110,6 +110,25 @@ def clean_up(storage_id: str, storage_manager: storage.S3StorageManager) -> None
     assert storage_id not in os.listdir(storage_manager._base_path)
 
 
+def test_delete_checkpoint(manager: storage.SharedFSStorageManager) -> None:
```

Comment:
This should go in `util.run_storage_lifecycle_test()` so that all storage managers automatically test for this behavior.

Reply:
(the current customer might only care about this backend, but if we feel it's important for one backend to work this way, it's likely valuable for all of them to work this way)